ratis-server/src/main/java/org/apache/ratis/server/impl/RaftConfiguration.java

```diff
@@ -206,6 +206,12 @@ boolean hasMajority(Collection<RaftPeerId> others, RaftPeerId selfId) {
         (oldConf == null || oldConf.hasMajority(others, selfId));
   }
 
+  /** @return true if the rejects are in the half(impossible for others in majority). */
+  boolean majorityImpossibleWithout(Collection<RaftPeerId> rejects) {
+    return conf.majorityImpossibleWithout(rejects) &&
+            (oldConf == null || oldConf.majorityImpossibleWithout(rejects));
+  }
```

Comment:
I think this check it not totally reverse  of hasMajority.  The reverse of hasMajority is `!conf.hasMajority(others, selfId) || !(oldConf == null || oldConf.hasMajority(others, selfId))` i.e.` !conf.hasMajority(others, selfId) || (oldConf != null && !oldConf.hasMajority(others, selfId))``. i.e. ` conf.majorityImpossibleWithout(rejects) || (oldConf != null && !oldConf.majorityImpossibleWithout(rejects))``

Reply:
This is right! I just forgot this.